src/fileimpl.cpp

```diff
@@ -121,6 +121,19 @@ namespace zim
         log_fatal("last offset (" << lastOffset << ") larger than file size (" << zimFile->fsize() << ')');
         throw ZimFileFormatError("last cluster offset larger than file size; file corrupt");
       }
+
+      std::call_once ( orderOnceFlag, [this]
+      {
+          auto nb_articles = getCountArticles().v;
+          articleListByCluster.reserve(nb_articles);
+
+          for(zim::article_index_type i = 0; i < nb_articles; i++)
+          {
+              articleListByCluster.push_back(std::make_pair(getDirent(article_index_t(i))->getClusterNumber().v, i));
+          }
+
+          std::sort(articleListByCluster.begin(), articleListByCluster.end());
+      });
```

Comment:
We want to call it once, but only if we need it.
Most applications (kiwix-serve, kiwix-desktop) will never loop over the articles. Generating this list all time we open a zim file is useless (and takes time).

We should call it in `findxByClusterOrder`